File: 2373. Largest Local Values in a Matrix.go
func largestLocal(grid [][]int) [][]int {
	n := len(grid)
	answer := make([][]int, n-2)
	for i := range n - 2 {
		answer[i] = make([]int, n-2)
		for j := range n - 2 {
			answer[i][j] = grid[i][j]
			for x := range 3 {
				for y := range 3 {
					answer[i][j] = max(answer[i][j], grid[i+x][j+y])
				}
			}
		}
	}
	return answer
}

File: python/2373. Largest Local Values in a Matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        answer = []
        for i in range(n-2):
            answer.append([])
            for j in range(n-2):
                val = grid[i][j]
                for x in range(3):
                    for y in range(3):
                        val = max(val, grid[i+x][j+y])
                answer[i].append(val)
        return answer
